instance-scheduler/main_int_test.go

```diff
@@ -13,7 +13,7 @@ func TestHandler(t *testing.T) {
 	t.Run("Test request", func(t *testing.T) {
 		// Accounts mi-platform-development and analytical-platform-data-development cause the main_int_test.go to fail because they are non-member accounts
 		// lacking the InstanceSchedulerAccess role, but they have the '-development' suffix typically present in member accounts.
-		os.Setenv("INSTANCE_SCHEDULING_SKIP_ACCOUNTS", "mi-platform-development,analytical-platform-data-development,analytical-platform-development,")
+		os.Setenv("INSTANCE_SCHEDULING_SKIP_ACCOUNTS", "mi-platform-development,analytical-platform-data-development,analytical-platform-development,moj-network-operations-centre-preproduction,")
```

Comment:
I don't believe that we should be coupling our continuous integration with external dependencies, which leaves the process fairly fragile. IMO best to remove this test completely from PR checks 👀

Reply:
Also not sure if I should add the environment the [template.yml ](https://github.com/ministryofjustice/modernisation-platform-instance-scheduler/blob/be0f617b33f88fab93aa8c555fa87267225fc4af/template.yaml#L39) file. Looking at the logs of the runs, it seems to skip the account and recognised it as a non-member account

Happy to add it for completeness though 😁

Reply:
This is a non-member preproduction account and according to the logic:
1. In general preproduction accounts should be included in the instance scheduling
2. however, not the non-member accounts and therefore such account should be indeed added to the [skip account env var](https://github.com/ministryofjustice/modernisation-platform-instance-scheduler/blob/be0f617b33f88fab93aa8c555fa87267225fc4af/template.yaml#L39).
This is not ideal that these accounts are hardcoded and perhaps this could be improved.

Regarding your comment on the value of the test. This is not a unit test, hence it is touching the API.
If I remember right, we decided on having both unit and integration tests, as if the test returned no instances or too many instances, we would know that something is properly broken. e.g. a role (where you cannot confirm that with unit tests). This tests are not calling start/stop API, but only retrieves instances info and counts their number (so it runs in test mode). We could revisit whether this integration test can be improved though.